nengo_loihi/hardware/nxsdk_shim.py

```diff
@@ -44,57 +60,67 @@ def __init__(self, *args, **kwargs):
                 src = f0.read()
                 dst = f1.read()
-                if src != dst:
-                    print("=== SOURCE: %s" % (cFilePath,))
+                if src != dst:  # pragma: no cover
+                    print("=== SOURCE: %s" % (c_file,))
                     print(src)
-                    print("\n=== DEST: %s" % (tmp_path,))
+                    print("\n=== DEST: %s" % (new_c_file,))
                     print(dst)
                     raise ValueError("Snip file not copied correctly")
 
-            # Also copy all the include files
-            include_path = os.path.join(tmp.name, name, "include")
-            shutil.copytree(includeDir, include_path)
-            assert os.path.isdir(include_path), "Copy failed %s" % include_path
-
-            return super().createProcess(name, tmp_path, include_path, *args, **kwargs)
-
-    d_set(snip_maker, b"R3JhcGg=", val=SnipMaker)
-
-except ImportError:
-    HAS_NXSDK = False
-    nxsdk = None
-    nxsdk_dir = None
-    nxsdk_version = None
+            new_include_dir = None
+            if include_dir is not None:
+                # Also copy all the include files
+                new_include_dir = os.path.join(tmp.name, name, "include")
+                shutil.copytree(include_dir, new_include_dir)
+                assert os.path.isdir(new_include_dir), (
+                    "Copy failed %s" % new_include_dir
+                )
 
-    exception = sys.exc_info()[1]
+            return new_c_file, new_include_dir
 
-    def assert_nxsdk(exception=exception):
-        raise exception
+        def createProcess(self, name, cFilePath, includeDir, *args, **kwargs):
+            # pragma: no cover (only used with older NxSDK)
+            cFilePath, includeDir = self._make_tmp_files(name, cFilePath, includeDir)
+            return super().createProcess(name, cFilePath, includeDir, *args, **kwargs)
+
+        def createSnip(self, phase, *args, **kwargs):
+            cppFile = kwargs.get("cFilePath", kwargs.get("cppFile", None))
+            includeDir = kwargs.get("includeDir", None)
+            name = kwargs.get("name", "generic_nengo_snip")
+            if cppFile is not None:
+                cppFile, includeDir = self._make_tmp_files(name, cppFile, includeDir)
+                kwargs["cFilePath" if "cFilePath" in kwargs else "cppFile"] = cppFile
+                if "includeDir" in kwargs:
+                    kwargs["includeDir"] = includeDir
+
+            return super().createSnip(phase, *args, **kwargs)
 
+    d_set(snip_maker, b"R3JhcGg=", val=SnipMaker)
 
-if HAS_NXSDK:
     micro_gen = d_import(b"bnhzZGsuY29tcGlsZXIubWljcm9jb2RlZ2VuLmludGVyZmFjZQ==")
     try:
         # try new location (nxsdk > 0.9.0)
         TraceConfigGenerator = d_import(
             b"bnhzZGsuYXJjaC5uMmEuY29tcGlsZXIudHJhY2VjZmdnZW4udHJhY2VjZmdnZW4=",
             b"VHJhY2VDZmdHZW4=",
         )
-    except ImportError:
+    except ImportError:  # pragma: no cover
```

Comment:
We only actively test against one NxSDK version at a time (i.e. the newest one we support). The difference between the 0.8.x versions were mostly these imports, so I think we should still support them with this compatibility code. But you're right, it would be good to check that.

That sounds like a good thing to do for the release (run with the oldest NxSDK we support as well as the newest) to make sure that everything still works. We can even run with coverage to make sure that all this compatibility code is still needed. I'll put a subtask on the release card.

Reply:
That said, I don't think there's a strong reason to support lots of older NxSDK versions. We could just drop support for all the 0.8.x versions; I think they're the main reasons we need all this compatibility code.